src/app/EventManagement.cpp

```diff
@@ -552,11 +553,17 @@ CHIP_ERROR EventManagement::CheckEventContext(EventLoadOutContext * eventLoadOut
 
     ReturnErrorOnFailure(ret);
 
+    DataModel::EventEntry eventInfo;
+    if (InteractionModelEngine::GetInstance()->GetDataModelProvider()->EventInfo(path, eventInfo) != CHIP_NO_ERROR)
+    {
+        eventInfo.readPrivilege = Access::Privilege::kView;
```

Comment:
Wait, why?  Why are we assuming this?

This seems to be inventing some non-spec behaviors, again.

But more importantly, shouldn't this error be fatal?  How do we have an event queued for a path we don't know anything about?

Reply:
What we have right now, in theory, is "everything has a privilege conceptually, but we only have things with non-default privileges in our data tables, and hence everything else has the default privileges".  Right now we have no real concept of "thing we implement that we don't have privileges for".

But what your new API has is "we will ask you for privileges and expect you to give them to us, but if you fail to do that we will guess it should be View".

Is callee allowed to just return error to get View?  Nothing documents that in the API.

Maybe I misunderstand the API contract?  But I would think the "default to view" can live down in the cluster impl (which you also have), and at _this_ level failure to get data should be failure.

Reply:
I could make the API as "cannot fail, you shall give me a privilege" but that feels odd (like you can ask for privileges on invalid paths like unknown clusters and you need to get something ... like the kView default).

Are you considering starting to have a CHIP_ERROR a good thing or not so good?
Code would be indeed simpler if we just guarantee you get something (and kView is the default just before).

Reply:
Updated code to treat event info failures as actual failures and not default to kView anymore.